.githooks/pre-commit

```diff
@@ -0,0 +1,22 @@
+echo "####### Chunks: Pre-commit hook #######"
+echo "[running] Static Check"
+if [[ ! `which staticcheck` ]]; then
+    echo "[FAILED] Missing staticcheck dependency - Install it from: https://staticcheck.io/docs/getting-started/"
+    exit 1
+else
+    staticcheck ./
+fi
+echo "[done] Static Check"
+echo "[done] Static Check"
+echo "[running] Go fmt"
+go fmt ./...
+echo "[done] Go fmt"
+echo "[running] Go test"
+go test --race .
+res=$?
```

Comment:
I thought staticcheck and gofmt weren't returning any other code than 0. Should we check all of them, then?

Reply:
Regarding `staticcheck`, it's `--help` reads:

```
  -fail checks
    	Comma-separated list of checks that can cause a non-zero exit status. (default "all")
```

So, yes, it returns a non-zero in case of errors. Also, this help shows nothing related to auto-correcting… so I think breaking the pipeline is expected.

Regarding `gofmt`, you can use `-w` so the auto-format feature corrects the files if needed.